### PR TITLE
[SPARK-7305][Streaming][WebUI] Make BatchPage show friendly information when jobs are dropped by SparkListener

### PR DESCRIPTION
**Pull request by zsxwing**
If jobs are dropped by SparkListener, at least we can show the job ids in BatchPage. Screenshot:

b1

### CODE REVIEW

**Comment by zsxwing** [comment]
cc @tdas

**Comment by zsxwing** [comment]
retest this please.

**Comment by zsxwing** [comment]
retest this please.

**Comment by zsxwing** [comment]
retest this please.

File: streaming/src/main/scala/org/apache/spark/streaming/ui/BatchPage.scala [hunk 1]
```diff
@@ -44,25 +45,33 @@ private[ui] class BatchPage(parent: StreamingTab) extends WebUIPage("batch") {
...
+        numSparkJobRowsInOutputOp, isFirstRow, sparkJob.sparkJobId)
+    }
+  }
+
   /**
    * Generate a row for a Spark Job. Because duplicated output op infos needs to be collapsed into
    * one cell, we use "rowspan" for the first row of a output op.
    */
-  def generateJobRow(
+  def generateNormalJobRow(
```
Comment (tdas) [inline streaming/src/main/scala/org/apache/spark/streaming/ui/BatchPage.scala]: Why is this not private?

Comment (zsxwing) [inline streaming/src/main/scala/org/apache/spark/streaming/ui/BatchPage.scala]: Sorry. It should be private.

File: streaming/src/main/scala/org/apache/spark/streaming/ui/BatchPage.scala [hunk 2]
```diff
@@ -122,27 +129,97 @@ private[ui] class BatchPage(parent: StreamingTab) extends WebUIPage("batch") {
...
+        generateJobRow(
+          outputOpId, description, formattedOutputOpDuration, sparkJobs.size, false, sparkJob)
       }.flatMap(x => x)
   }
 
+  private def generateOutputOpDescription(sparkJobs: Seq[SparkJobIdWithUIData]): Seq[Node] = {
+    val lastStageInfo =
+      sparkJobs.flatMap(_.jobUIData).headOption. // Get the first JobUIData
+        // Get the latest Stage info
+        flatMap { sparkJob =>
```
Comment (tdas) [inline streaming/src/main/scala/org/apache/spark/streaming/ui/BatchPage.scala]: nit: This is a little weird to read with the continuous line havnig a line of comment in the middle. took 5 seconds to realize that this is a continuation line.

**Comment by tdas** [comment]
LGTM. Will merge when tests pass.

**Comment by zsxwing** [comment]
retest this please

**Pull request closed without merging**